include/wlr/types/wlr_buffer.h

```diff
@@ -76,6 +78,14 @@ void wlr_buffer_unlock(struct wlr_buffer *buffer);
  */
 bool wlr_buffer_get_dmabuf(struct wlr_buffer *buffer,
 	struct wlr_dmabuf_attributes *attribs);
+/**
+ * Access a pointer to the allocated data from the underlying implementation,
+ * and its size.
+ *
+ * The returned pointer should be pointing to a valid memory location.
+ */
+bool wlr_buffer_get_data_ptr(struct wlr_buffer *buffer, void **data,
```

Comment:
Can we move this into an internal header for now? (e.g. `include/types/wlr_buffer.h`)

I'm not yet 100% sure this is the right interface, not making it public right away allows us to more easily change it if necessary.

Reply:
Sure